src/exec-child.js

```diff
@@ -18,17 +18,22 @@ var execOptions = params.execOptions;
 var pipe = params.pipe;
 var stdoutFile = params.stdoutFile;
 var stderrFile = params.stderrFile;
-var codeFile = params.codeFile;
-
-var c = childProcess.exec(cmd, execOptions, function (err) {
-  if (!err) {
-    fs.writeFileSync(codeFile, '0');
-  } else if (err.code === undefined) {
-    fs.writeFileSync(codeFile, '1');
-  } else {
-    fs.writeFileSync(codeFile, err.code.toString());
-  }
-});
+
+var c;
+try {
+  c = childProcess.exec(cmd, execOptions, function (err) {
+    if (!err) {
+      process.exit(0);
+    } else if (err.code === undefined) {
+      process.exit(1);
+    } else {
+      process.exit(err.code);
+    }
+  });
+} catch (e) {
+  // child_process could not run the command.
+  process.exit(127);
```

Comment:
`process.exit` is potentially a bad idea if there are still callbacks in the queue, or if there are async tasks running, because they will not finish. Of course, at this point, neither of these should be the case, but I still think setting `process.exitCode` is a safer way to do this.

Reply:
Also, below this, we are manually ending `stdoutStream` and `stderrStream` when `c.stdout` and `c.stderr` end. Why do they need to be ended manually?

Reply:
Ack on `process.exitCode`, sounds like it will be safer.

I'm not sure if we need to manually end the stream. This is legacy code. What would end the streams otherwise, though?

Reply:
I would need to test this to make sure, but just a normal `pipe` should work:
```
c.stdout.pipe(stdoutStream);
c.stderr.pipe(stderrStream);
```

Reply:
I don't think we can do that. We need things as-they-are in order to:

1. Send  output to stdio in real-time
2. Save that output in a file (for when `exec-child.js` finishes)

If we pipe the output, then we can't achieve bullet 2.

Reply:
You can just pipe twice:
```
c.stdout.pipe(fs.createWriteStream(stdoutFile));
c.stderr.pipe(fs.createWriteStream(stderrFile));
c.stdout.pipe(process.stdout);
c.stderr.pipe(process.stderr);
```

Reply:
Nevermind, I was confused. You're right 👍